src/web-auth/captcha.js

```diff
@@ -95,55 +108,88 @@ function scriptForCaptchaProvider(provider, lang, callback) {
       );
     case HCAPTCHA_PROVIDER:
       return (
-        'https://js.hcaptcha.com/1/api.js?hl=' +
-        lang +
-        '&onload=' +
-        callback
+        'https://js.hcaptcha.com/1/api.js?hl=' + lang + '&onload=' + callback
       );
     case FRIENDLY_CAPTCHA_PROVIDER:
       return 'https://cdn.jsdelivr.net/npm/friendly-challenge@0.9.12/widget.min.js';
+    case ARKOSE_PROVIDER:
+      return (
+        'https://' +
+        clientSubdomain +
+        '.arkoselabs.com/v2/' +
+        siteKey +
+        '/api.js'
+      );
     /* istanbul ignore next */
     default:
       throw new Error('Unknown captcha provider');
   }
 }
 
-function injectCaptchaScript(element, opts, callback) {
-  var providerName;
-  switch (opts.provider) {
-    case RECAPTCHA_ENTERPRISE_PROVIDER:
-      providerName = 'recaptcha';
-      break;
-    case RECAPTCHA_V2_PROVIDER:
-      providerName = 'recaptcha';
-      break;
-    case HCAPTCHA_PROVIDER:
-      providerName = 'hcaptcha';
-      break;
-    case FRIENDLY_CAPTCHA_PROVIDER:
-      providerName = 'friendly_captcha';
-      break;
+function loadScript(url, attributes) {
+  var script = window.document.createElement('script');
+  for (var attr in attributes) {
+    if (attr.startsWith('data-')) {
+      script.dataset[attr.replace('data-', '')] = attributes[attr];
+    } else {
+      script[attr] = attributes[attr];
+    }
   }
-  var callbackName = providerName + 'Callback_' + Math.floor(Math.random() * 1000001);
-  window[callbackName] = function() {
-    delete window[callbackName];
-    callback();
+  script.src = url;
+  window.document.body.appendChild(script);
+}
+
+function removeScript(url) {
+  var scripts = window.document.querySelectorAll('script[src="' + url + '"]');
+  scripts.forEach(function (script) {
+    script.remove();
+  });
+}
+
+function injectCaptchaScript(element, opts, callback, setValue) {
+  var callbackName =
+    opts.provider + 'Callback_' + Math.floor(Math.random() * 1000001);
```

Comment:
Maybe I am missing it in all the changes, but isnt this changing the callbackName for providers such as `RECAPTCHA_V2_PROVIDER` ? `opts.provider` in that case is equal to `recaptcha_v2`, while we used to set the providerName to `recaptcha` in the switch case.

Same with `RECAPTCHA_ENTERPRISE_PROVIDER`.

If it is, can we verify this is expected?

Reply:
Yes you're right. Do you see any issues with changing it? From my perspective, it's arbitrary and I just removed the existing switch to make the code cleaner.

Reply:
I have no idea whether or not it matters. If you are sure it has no implications, all fine for me.

Reply:
The only thing that comes to mind is that if some user is writing scripts against this captcha callback name (would be unlikely). But again, I don't think that would be good practise and we probably shouldn't be responsible if this name change causes them issues. From our side, the exact name is not important.